engine/model_draft_transactions.go

```diff
@@ -164,6 +165,14 @@ func (m *DraftTransaction) processConfigOutputs(ctx context.Context) error {
 	conditions := map[string]interface{}{
 		xPubIDField: m.XpubID,
 	}
+
+	// Get the sender's paymail from the metadata, this help when sender has multiple paymails
+	senderPaymail, ok := m.Metadata["sender"].(string)
+	if ok {
+		alias, _, _ := paymail.SanitizePaymail(senderPaymail)
+		conditions["alias"] = alias
+	}
```

Comment:
This is the function implementation from `paymail`:
```golang
func SanitizePaymail(paymailAddress string) (alias, domain, address string) {

	// Sanitize the paymail address
	address = SanitizeEmail(paymailAddress)

	// Split the email parts (alias @ domain)
	parts := strings.Split(address, "@")

	// Sanitize the domain name (force to lowercase, remove www.)
	if len(parts) > 1 {
		domain, _ = SanitizeDomain(parts[1])
	}

	// Set the alias (lowercase, no spaces)
	alias = strings.TrimSpace(strings.ToLower(parts[0]))

	// Paymail address does not meet the basic requirement of an email address
	// Since we don't return an error, we will return an empty result
	if len(alias) == 0 || len(domain) == 0 {
		address = ""
	}

	return
}
```

I think, we should check the last "address" returned value according to the comment: `"Since we don't return an error, we will return an empty result"`.


```suggestion
	if ok {
		alias, _, address := paymail.SanitizePaymail(senderPaymail)
		if address != "" {
		   conditions["alias"] = alias
		}
	}
```

Reply:
Done